src/NServiceBus.Persistence.AzureStorage.ComponentTests/Sagas/When_working_with_saga_with_correlation_property.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace NServiceBus.Persistence.AzureStorage.ComponentTests.Persisters
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Extensibility;
+    using NServiceBus.Sagas;
+    using NUnit.Framework;
+
+    public class When_working_with_saga_with_correlation_property
+    {
+        const string CorrelationValue = "correlation-value";
+        const string UpdatedValue = "updated-value";
+
+        [Test]
+        public async Task Should_not_issue_table_scan()
+        {
+            var connectionString = AzurePersistenceTests.GetConnectionString();
+
+            // warm up table cache
+            var warmUp = new AzureSagaPersister(connectionString, true);
+            await warmUp.Get<SagaData>(Guid.NewGuid(), null, new ContextBag()).ConfigureAwait(false);
+
+            using (var recorder = new AzureRequestRecorder())
+            {
+                // save saga
+                {
+                    var saga = new SagaData
+                    {
+                        Id = Guid.NewGuid(),
+                        Originator = "Moo",
+                        OriginalMessageId = "MooId",
+                        Correlation = CorrelationValue
+                    };
+
+                    var persister = new AzureSagaPersister(connectionString, true);
+                    await persister.Save(saga, new SagaCorrelationProperty(nameof(SagaData.Correlation), CorrelationValue), null, new ContextBag()).ConfigureAwait(false);
+                }
+
+                // get by correlation and update
+                {
+                    var persister = new AzureSagaPersister(connectionString, true);
+                    var ctx = new ContextBag();
+
+                    var saga = await persister.Get<SagaData>(nameof(SagaData.Correlation), CorrelationValue, null, ctx).ConfigureAwait(false);
+                    saga.Value = UpdatedValue;
+
+                    await persister.Update(saga, null, ctx).ConfigureAwait(false);
+                }
+
+                // get by correlation and complete
+                {
+                    var persister = new AzureSagaPersister(connectionString, true);
+                    var ctx = new ContextBag();
+
+                    var saga = await persister.Get<SagaData>(nameof(SagaData.Correlation), CorrelationValue, null, ctx).ConfigureAwait(false);
+                    await persister.Complete(saga, null, ctx).ConfigureAwait(false);
+                }
+
+                recorder.Print(Console.Out);
+
+                var gets = recorder.Requests.Where(r => r.ToLower().Contains("get"));
+                var getsWithNoPartitionKey = gets.Where(get => get.Contains("PartitionKey") == false).ToArray();
```

Comment:
There is a bug in the assertion of the test.  The assertion is checking for any GET requests that to not have the string `PartitionKey` in them.  Since the query that ends up doing a table scan is explicitly selecting `PartitionKey` and `RowKey` fields, the `PartitionKey` string is present in the request URL, even though it is not in the query.  Perhaps a better approach would be to parse the URLs and only look at the $filter, but a quick and dirty approach I took to make the test fail was to explicitly replace `select=PartitionKey` with `String.Empty` before the assertion.  Here is my modified code:

```csharp
var getsWithNoPartitionKey = gets.Where(get => get.Replace("select=PartitionKey", String.Empty).Contains("PartitionKey") == false).ToArray();
```